symfony/plugins/orangehrmMarketPlacePlugin/modules/marketPlace/actions/installAddonAPIAction.class.php

```diff
@@ -57,28 +57,36 @@ public function execute($request)
     /**
      * @param $addonURL
      * @param $addonDetail
+     * @return int
      * @throws CoreServiceException
      * @throws DaoException
      */
     private function getAddonFile($addonURL, $addonDetail)
     {
-        $addonfile = $this->getApiManagerService()->getAddonFile($addonURL);
-        return $this->installAddon($addonfile, $addonDetail);
+        $addonFilePath = $this->getApiManagerService()->getAddonFile($addonURL);
+        return $this->installAddon($addonFilePath, $addonDetail);
     }
 
     /**
-     * @param $addon
+     * @param $addonFilePath
      * @param $addonDetail
+     * @return int
      * @throws DaoException
      */
-    protected function installAddon($addon, $addonDetail)
+    protected function installAddon($addonFilePath, $addonDetail)
     {
+        $pluginname = $this->getMarcketplaceService()->extractAddonFile($addonFilePath);
+        $currentDirectory = dirname(__FILE__);
+        chdir($currentDirectory . "/../../../../../");
+        exec("php symfony cc", $symfonyCcResponse, $status);
+        if ($status !== 0) {
+        /////////////////////
```

Comment:
here I need to return a error code. and say error occur. need help on that since Nuwan said to return a exception code.